common/djangoapps/xblock_django/user_service.py

```diff
@@ -68,6 +73,27 @@ def get_anonymous_user_id(self, username, course_id):
         course_id = CourseKey.from_string(course_id)
         return anonymous_id_for_user(user=user, course_id=course_id)
 
+    def get_current_user_id(self):
```

Comment:
I added these methods for convenience, but it means that ModuleSystem relies on methods that aren't declared on the reference XBlock UserService.

Should these be added there as well (and raise UnimplementedError)?

Reply:
@symbolist Are you sure you don't want me to re-add these helper methods? The refactoring is pretty verbose without them: e029feb46c0a74d7eb89a7717382663486c78044

Reply:
I think that is only because of using the `get_anonymous_user_id()` method for getting the anonymous id? And XBlocks outside this repo are not going to be importing the keys from edx-platform. They will directly use the value of those keys. So if we follow the intent of the XBlockUser [interface](https://github.com/edx/XBlock/blob/master/xblock/reference/user_service.py#L23-L51), all the info related to a user should be an attribute of that object and can be accessed via `user.<attribute>` for required attributes and `user.opt_attrs['attribute']` for optional ones.

Reply:
Ah, that's much nicer :) Addressed this with ad0a247 .